### PR TITLE
fix: Modify region list to a boto3 call

### PR DESCRIPTION
**Pull request by quixoticmonk**
**Issue number:#25**

### Changes

The goal of this PR is update the regions list  maintained as a hardcoded list to an api call which pulls the active supported regions.

### User experience

No Changes.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

- [X] I have performed a self-review of this change
- [X] Changes have been tested
- [ ] Changes are documented

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: eksupgrade/cli.py [hunk 1]
```diff
@@ -103,5 +79,15 @@ def entry(args: Optional[List[str]] = None) -> None:
     main(parsed_arguments)
 
 
+def get_eks_supported_regions() -> List[str]:
+    """Retrieve the active regions supporting EKS across aws and us-gov partitions"""
+    session = boto3.session.Session()
+    partition_list: List[str] = ["aws", "aws-us-gov"]
```
Comment (bryantbiggs) [inline eksupgrade/cli.py:85]: we should also include `"aws-cn"`

File: eksupgrade/cli.py [hunk 2]
```diff
@@ -103,5 +79,15 @@ def entry(args: Optional[List[str]] = None) -> None:
     main(parsed_arguments)
 
 
+def get_eks_supported_regions() -> List[str]:
```
Comment (mbeacom) [inline eksupgrade/cli.py:82]: I thought about doing this myself when I expanded the list, but I am not sure we even want this region enforcement.  We don't actually use the region anywhere except maybe the unused eksctl module (which should likely be removed).

@bryantbiggs Thoughts?  Should we simply eliminate this all together?

Comment (bryantbiggs) [reply eksupgrade/cli.py:82]: ya, I don't see a valid use case where we need to know the supported regions. Users are going to tell us their cluster name and the region its located in - if they do either of those incorrectly, its just incorrect and they will need to re-provide the correct information

If we were trying to create a new cluster from scratch - then supported regions would have a valid use case (maybe, the API would still return an error if the region is not supported)

Comment (quixoticmonk) [reply eksupgrade/cli.py:82]: That makes sense from the api call and plan forward. We can close the issue # [25](https://github.com/aws-samples/amazon-eks-one-click-cluster-upgrade/issues/25) and this PR then.

Comment (mbeacom) [reply eksupgrade/cli.py:82]: Thank you!

**Pull request closed without merging**